docs/root/configuration/listeners/network_filters/tcp_proxy_filter.rst

```diff
@@ -17,6 +17,105 @@ other network filters on a per-connection basis by setting a per-connection
 state object under the key `envoy.tcp_proxy.cluster`. See the
 implementation for the details.
 
+.. _config_network_filters_tcp_proxy_subset_lb:
+
+Routing to a subset of hosts
+----------------------------
+
+TCP proxy can be configured to route to a subset of hosts within an upstream cluster.
+
+To define metadata that a suitable upstream host must match, use one of the following fields:
+
+1. Use :ref:`TcpProxy.metadata_match<envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.metadata_match>`
+to define required metadata for a single upstream cluster.
+2. Use :ref:`ClusterWeight.metadata_match<envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.WeightedCluster.ClusterWeight.metadata_match>`
+to define required metadata for a weighted upstream cluster.
+3. Use combination of :ref:`TcpProxy.metadata_match<envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.metadata_match>`
+and :ref:`ClusterWeight.metadata_match<envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.WeightedCluster.ClusterWeight.metadata_match>`
+to define required metadata for a weighted upstream cluster (matadata from the latter will be merged on top of the former).
+
+E.g.,
```

Comment:
I think we can leave off the example blocks. Inline examples like this have a habit of becoming stale and I think you've done a good job of explaining how to do this, so they're not really necessary.

Reply:
done